shared/actions/fs.js

```diff
@@ -212,29 +212,35 @@ function* pollSyncStatusUntilDone(): Saga.SagaGenerator<any, any> {
     return
   }
   polling = true
+  let syncingSet = false
   try {
     let status: RPCTypes.FSSyncStatus = yield Saga.call(RPCTypes.SimpleFSSimpleFSSyncStatusRpcPromise)
-    let _kbfsUploading = false // Don't send duplicates else we get high cpu usage.
+    if (status.totalSyncingBytes <= 0) {
+      return
+    }
+
+    yield Saga.put(NotificationsGen.createBadgeApp({key: 'kbfsUploading', on: true}))
+    yield Saga.put(FsGen.createSetFlags({syncing: true}))
+    syncingSet = true
+
     while (status.totalSyncingBytes > 0) {
-      if (!_kbfsUploading) {
-        _kbfsUploading = true
-        yield Saga.put(NotificationsGen.createBadgeApp({key: 'kbfsUploading', on: true}))
-        yield Saga.put(FsGen.createSyncingStatus({isSyncing: true}))
-      }
       yield Saga.delay(2000)
       status = yield Saga.call(RPCTypes.SimpleFSSimpleFSSyncStatusRpcPromise)
     }
-    if (_kbfsUploading) {
+
+    yield Saga.put(NotificationsGen.createBadgeApp({key: 'kbfsUploading', on: false}))
+    yield Saga.put(FsGen.createSetFlags({syncing: false}))
+    syncingSet = false
+  } finally {
+    if (syncingSet) {
```

Comment:
Instead of making a variable, you can do the following here:
```javascript
const syncingSet = yield Saga.select((state) => state.fs.flags.syncing)
```